### PR TITLE
Add total cores to linux cpu plugin

### PR DESCRIPTION
**Pull request by sh9189**
Add a cores attributes to get total number of cores on linux machines.

### CODE REVIEW

File: lib/ohai/plugins/linux/cpu.rb [hunk 1]
```diff
@@ -80,5 +80,6 @@
     cpu cpuinfo
     cpu[:total] = cpu_number
     cpu[:real] = real_cpu.keys.length
+    cpu[:cores] = real_cpu.keys.length * cpu["0"]["cores"].to_i
```
Comment (mcquin) [inline lib/ohai/plugins/linux/cpu.rb]: Are all cpus guaranteed to have the same number of cores?

Comment (sh9189) [inline lib/ohai/plugins/linux/cpu.rb]: I believe so. At least that's what I see on all of our Linux machines.

**Comment by thommay** [comment]
i think this is fine, :+1:

File: lib/ohai/plugins/linux/cpu.rb (hunk 1 above)
Comment (lamont-granquist) [inline lib/ohai/plugins/linux/cpu.rb]: Yeah I think its fairly unlikely for a Linux box to have a mix of say dual core and quad core in different sockets.  That may, in fact, make lots of MBs sad if the CPUs aren't reasonably identicalish.  Certainly, we can fix it if it ever becomes a problem, YAGNI.

**Comment by lamont-granquist** [comment]
:+1:

**Pull request merged**